src/simularium/versionHandlers.ts

```diff
@@ -1,27 +1,94 @@
+import { mapValues } from "lodash";
 import * as si from "si-prefix";
 
 import { DEFAULT_CAMERA_SPEC } from "../constants";
 import {
+    CubeDisplayType,
+    GizmoDisplayType,
+    ObjDisplayType,
+    PdbDisplayType,
+    SphereDisplayType,
     TrajectoryFileInfo,
     TrajectoryFileInfoAny,
     TrajectoryFileInfoV1,
 } from "./types";
+import { AgentDisplayDataWithGeometry } from "./VisGeometry";
 
+// the data may come in missing any of these values
+export interface AgentTypeVisDataPreProcessing {
+    displayType?:
+        | PdbDisplayType
+        | ObjDisplayType
+        | SphereDisplayType
+        | CubeDisplayType
+        | GizmoDisplayType;
+    url?: string;
+    color?: string;
+}
+
+export interface AgentDisplayDataPreProcessing {
+    name: string;
+    geometry?: AgentTypeVisDataPreProcessing;
+}
+export interface EncodedTypeMappingPreProcessing {
+    [key: number]: AgentDisplayDataPreProcessing;
+}
 /*
 Handles different trajectory file format versions.
 Currently supported versions: 1, 2
 */
-const LATEST_VERSION = 2;
+const LATEST_VERSION = 3;
 const VERSION_NUM_ERROR = "Invalid version number in TrajectoryFileInfo:";
 
+export const sanitizeAgentMapGeometryData = (
+    typeMapping: EncodedTypeMappingPreProcessing
+): { [key: number]: AgentDisplayDataWithGeometry } => {
+    return mapValues(typeMapping, (value: AgentDisplayDataPreProcessing) => {
+        let geometry = {};
+        if (value.geometry) {
+            let url = value.geometry.url || "";
+            let { displayType } = value.geometry;
+            if (!displayType) {
+                // we're relying on the data to have a displayType to tell us what sort of data the url is pointing at
+                // if the user fails to provide the displayType, we'll default to loading a sphere, and clear out the url
+                url = "";
+                displayType = "SPHERE";
+            }
+
+            geometry = {
+                ...value.geometry,
+                displayType,
+                url,
+                color: value.geometry.color || "",
+            };
+        } else {
+            geometry = {
+                displayType: "SPHERE",
+                url: "",
+                color: "",
+            };
+        }
+        return {
+            ...value,
+            geometry,
+        } as AgentDisplayDataWithGeometry;
+    });
+};
+
 export const updateTrajectoryFileInfoFormat = (
     msg: TrajectoryFileInfoAny
 ): TrajectoryFileInfo => {
-    let output = msg;
+    let output = {
+        ...msg,
+        typeMapping: sanitizeAgentMapGeometryData(msg.typeMapping),
+        version: LATEST_VERSION,
+    };
 
     switch (msg.version) {
         case LATEST_VERSION:
             break;
+        case 2:
+            break;
```

Comment:
is there any distinction between v2 and v3 now?  I didn't see any definitions for TrajectoryFileInfoV2 vs V3

Reply:
currently no, but I was going to bring in the other changes for v3 in a different PR. So far all the v3 changes are optional.